### PR TITLE
Make use of nameOf for naming telemetry traces in participant-integration-api

### PR DESCRIPTION
**Pull request by realvictorprm**
This replaces the hardcoded fully qualified names for the method traces via usage of `qualifiedNameOfCurrentFunction` definied in a new scala macro lib called `nameof` within this repo.

Nothing should change for the user.

Side note: I've made this change in the process of figuring out how the telemetry api is working to implement it for other projects within this repository too.

~Technical side note:
I'm not satisfied with this `nameof` solution because the construction of the fqn can be done with a slightly adjusted implementation of the macro from the library. I tried to cut that macro out into a small lib within this repo but I failed to get the build system correctly working with a library providing the macro.~

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by stefanobaghino-da — COMMENTED** [review]
Love it!

I'll leave the approval to the team in charge of the indexer and integration API to make sure they're up to speed with this change. 😉

**Comment by hubert-da** [comment]
I took a glance at your PR yd (and today too) and forgot to ask one thing. Have you seen the stuff we have under the metrics package, like `TelemetryContext`? Do you think we could somehow unify this? Anyway, the changes look fine to me, as long as there is no impact of using reflection.

**Review by hubert-da — APPROVED** [review]
I took a glance at your PR yd (and today too) and forgot to ask one thing. Have you seen the stuff we have under the metrics package, like TelemetryContext? Do you think we could somehow unify this? Anyway, the changes look fine to me.

**Comment by realvictorprm** [comment]
As I'm currently anyways unifying contexts from Logging, Tracing and Metrics I'll take a look @hubert-da :)

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala [hunk 1]
```diff
@@ -4,7 +4,6 @@
 package com.daml.platform.store.appendonlydao.events
 
 import java.sql.Connection
-
```
Comment (realvictorprm) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:7]: I actually did not intend to remove spaces :|

Comment (realvictorprm) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:7]: or newlines, whatever :smile:

Comment (realvictorprm) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala]: Now we have that newline again :)

**Pull request merged**